src/NuGet.Configuration/Utility/SettingsUtility.cs

```diff
@@ -130,16 +130,16 @@ public static string GetGlobalPackagesFolder(ISettings settings)
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            // Note that the value GlobalPackagesFolder must be a full path and not a relative path
-            // Read it like a string and explicitly not as a path
-            var path = settings.GetValue(ConfigSection, GlobalPackagesFolderKey, isPath: false);
+            var path = Environment.GetEnvironmentVariable(GlobalPackagesFolderEnvironmentKey);
             if (!string.IsNullOrEmpty(path))
             {
                 path = path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
```

Comment:
We need to replace it. We are reading in the value as a string, not as a path. Note that we have always done this for repositoryPath